examples/module/mpc/cartpole.py

```diff
@@ -0,0 +1,153 @@
+    g = 9.81
+    time  = torch.arange(0, T, device=device) * dt
+
+    expert = dict(
+        Q = torch.tile(torch.eye(n_state + n_ctrl, device=device), (n_batch, T, 1, 1)),
+        p = torch.randn(n_batch, T, n_sc).to(device),
+        len = torch.tensor(1.5).to(device),
+        m_cart = torch.tensor(20).to(device),
+        m_pole = torch.tensor(10).to(device),
+    )
+    fname = os.path.join(args.save, 'expert.pkl')
+    with open(fname, 'wb') as fi:
+        pkl.dump(expert, fi)
+
+    current_x = torch.zeros(1, T, n_state, device=device)
+    current_u = torch.sin(time).unsqueeze(1).unsqueeze(0)
+    x_init = torch.tensor([[0, 0, torch.pi*0.5, 0]], device=device)
+    current_x[...,0,:] = x_init
+    expert_cartPoleSolver = CartPole(dt, expert['len'], expert['m_cart'], expert['m_pole'], g).to(device)
+    for i in range(T-1):
+        current_x[...,i+1,:], _ = expert_cartPoleSolver(current_x[...,i,:], current_u[...,i,:])
+
+    torch.manual_seed(args.seed)
+    len = torch.tensor(1.55).to(device).requires_grad_()
+    m_cart = torch.tensor(20.3).to(device).requires_grad_()
+    m_pole = torch.tensor(10.4).to(device).requires_grad_()
+
+    fname = os.path.join(args.save, 'cartpole losses.csv')
+    loss_f = open(fname, 'w')
+    loss_f.write('im_loss,mse\n')
+    loss_f.flush()
+
+    mpc_expert = pp.module.MPC(expert_cartPoleSolver, T, step=15).to(device)
+    x_true, u_true, cost_true = mpc_expert(expert['Q'], expert['p'], x_init, current_x, current_u, time)
+
+    def get_loss(_len, _m_cart, _m_pole):
+        current_x = torch.zeros(1, T, n_state, device=device)
+        current_u = torch.sin(time).unsqueeze(1).unsqueeze(0)
+        x_init = torch.tensor([[0, 0, torch.pi*0.5, 0]], device=device)
+        current_x[...,0,:] = x_init
+        agent_cartPoleSolver = CartPole(dt, _len, _m_cart, _m_pole, g).to(device)
+        for i in range(T-1):
+            current_x[...,i+1,:], _ = agent_cartPoleSolver(current_x[...,i,:], current_u[...,i,:])
+
+        mpc_agent = pp.module.MPC(agent_cartPoleSolver, T, step=15).to(device)
+        x_pred, u_pred, cost_pred = mpc_agent(expert['Q'], expert['p'], x_init, current_x, current_u, time)
+
+        traj_loss = torch.mean((u_true - u_pred)**2) \
+            + torch.mean((x_true - x_pred)**2)
+
+        return traj_loss
+
+    opt = optim.RMSprop((len, m_cart, m_pole), lr=1e-2)
+
+    for i in range(100):
+        traj_loss = get_loss(len, m_cart, m_pole)
+        opt.zero_grad()
+        traj_loss.backward()
+        opt.step()
+
+        model_loss = torch.mean((len - expert['len'])**2) + \
+                     torch.mean((m_cart - expert['m_cart'])**2) + \
+                     torch.mean((m_pole - expert['m_pole'])**2)
+
+        loss_f.write('{},{}\n'.format(traj_loss.item(), model_loss.item()))
+        loss_f.flush()
+
+        plot_interval = 1
+        if i % plot_interval == 0:
+            os.system('./plot.py "{}" &'.format(args.save))
+            print(len, expert['len'])
```

Comment:
Can we print some info which are easy for the user? Let's say these length and masses of expert and agent so that user can easily see what are you printing.